atomic_reactor/util.py

```diff
@@ -320,14 +323,27 @@ def clone_git_repo(git_url, target_dir, commit=None, retry_times=GIT_MAX_RETRIES
     logger.debug("url = '%s', dir = '%s', commit = '%s'",
                  git_url, target_dir, commit)
 
-    cmd = ["git", "clone", git_url, quote(target_dir)]
+    cmd = ["git", "clone"]
+    if branch:
+        cmd += ["-b", branch, "--single-branch"]
+    if depth:
+        cmd += ["--depth", '{0}'.format(depth)]
+    cmd += [git_url, quote(target_dir)]
 
     logger.debug("cloning '%s'", cmd)
+    repo_commit = ''
     for counter in range(retry_times + 1):
         try:
             # we are using check_output, even though we aren't using
             # the return value, but we will get 'output' in exception
             subprocess.check_output(cmd, stderr=subprocess.STDOUT)
+            repo_commit = reset_git_repo(target_dir, commit)
+            if depth and repo_commit != commit:
+                cmd = ["git", "fetch", "--unshallow"]
+                subprocess.check_output(cmd, stderr=subprocess.STDOUT)
+                repo_commit = reset_git_repo(target_dir, commit)
+                if repo_commit != commit:
+                    continue
```

Comment:
These last two lines look wrong. If we still don't have the commit after fetching the whole lot, we'll get an exception from `reset_git_repo` and should give up immediately.

Reply:
Redid the logic using --deepen.